### DISCUSSION TITLE
How to initiate disconnect of a debug session rather than a stop

### DISCUSSION THREAD

**Discussion by JustinGrote** [microsoft/vscode-discussions#767] [Extension Development QnA] [answered]
In the PowerShell extension, we start a child debug session from the C# extension using the "attach" launch config. In the debug UI, there is an option for "stop" vs "disconnect" where stop kills the process and the disconnect leaves it running.

In the Debug session API, it appears to only have a [stopDebugging()](https://github.com/microsoft/vscode/blob/2d143076200472649ed2756908f6aa79c75ad960/src/vscode-dts/vscode.d.ts#L15331) function. Is there an equivalent option for `disconnectDebugging()` or does this need to be a feature request? Is it possible to send the disconnect LSP command to the other extension directly?

**Comment by JustinGrote** [comment, marked as answer]
I ended up sending a `customRequest` as a workaround but hopefully this is a planned API feature?
https://github.com/PowerShell/vscode-powershell/pull/4704

**Comment by theReynald** [comment]
Hi Justin. You may consider making a feature request. Closing due to discovered work-around resolution.
